config/initializers/rswag_api.rb

```diff
@@ -10,5 +10,36 @@
   # The function will have access to the rack env for the current request
   # For example, you could leverage this to dynamically assign the "host" property
   #
-  #c.swagger_filter = lambda { |swagger, env| swagger['host'] = env['HTTP_HOST'] }
+  c.swagger_filter = lambda do |swagger, env|
+
+    swagger['host'] = env['HTTP_HOST']
+    http = swagger['host'].match(/localhost/) ? 'http' : 'https'
+
+
+    # Use variables to iterate and replace with values in the swagger file template.
+    variables = {
+      '$http' => http,
+      '$host' => swagger['host']
+    }
+
+    transform_values!(swagger, variables)
+  end
+
+  private
+
+  # This is a deep value replacement for a hash object.
+  def self.transform_values!(value, variables={})
+    if value.is_a? String
```

Comment:
Just an FYI about why this is needed, it's so that the example in the description refers to the current server that is running the api vs it being hardcoded to our k8 qa instance.

Reply:
I removed this for now in order not to complicate this PR and also  in the mean time look to see if I can take another approach.